crates/gitbutler-tauri/src/main.rs

```diff
@@ -246,11 +246,11 @@ fn main() {
                 .menu(menu::build)
                 .on_window_event(|window, event| match event {
                     #[cfg(target_os = "macos")]
-                    tauri::WindowEvent::CloseRequested { api, .. } => {
-                        if window.app_handle().windows().len() == 1 {
-                            tracing::debug!("Hiding all application windows and preventing exit");
-                            window.app_handle().hide().ok();
-                            api.prevent_close();
+                    tauri::WindowEvent::CloseRequested { .. } => {
+                        let app_handle = window.app_handle();
+                        if app_handle.windows().len() == 1 {
+                            app_handle.cleanup_before_exit();
+                            app_handle.exit(0);
                         }
```

Comment:
Is an else clause containing `api.prevent_close()` not required?

Reply:
My understanding of applications on macOs was that if there are two windows, it's still one instance of the app. As such, we want to prevent the killing of the applicaiton if there a window count that is not equal to 1?

Reply:
Right, that makes sense. I did, however, test this manually and with two windows open the you can close one without the app quitting.

Reply:
Cool. That sounds all good then. Thanks :+1: